From a1adea65049c1ceef7eda98360dc6455512120df Mon Sep 17 00:00:00 2001
From: Ma Wupeng <mawupeng1@huawei.com>
Date: Tue, 16 Aug 2022 15:37:21 +0800
Subject: [PATCH 018/261] [Huawei] memblock: Add static to memblock function
 releted to memory reliable

Offering: HULK
hulk inclusion
category: feature
bugzilla: 187270

--------------------------------

Fix static warnings related to memroy reliable.

Signed-off-by: Ma Wupeng <mawupeng1@huawei.com>
---
 mm/hugetlb.c  | 9 +++++----
 mm/memblock.c | 8 +++-----
 2 files changed, 8 insertions(+), 9 deletions(-)

diff --git a/mm/hugetlb.c b/mm/hugetlb.c
index 683e4e0c9018..a325faa376dd 100644
--- a/mm/hugetlb.c
+++ b/mm/hugetlb.c
@@ -2697,10 +2697,11 @@ struct page *alloc_huge_page(struct vm_area_struct *vma,
 	return ERR_PTR(-ENOSPC);
 }
 
-void *__init __alloc_bootmem_huge_page_inner(phys_addr_t size,
-					     phys_addr_t align,
-					     phys_addr_t min_addr,
-					     phys_addr_t max_addr, int nid)
+static void *__init __alloc_bootmem_huge_page_inner(phys_addr_t size,
+						    phys_addr_t align,
+						    phys_addr_t min_addr,
+						    phys_addr_t max_addr,
+						    int nid)
 {
 	if (!mem_reliable_is_enabled())
 		return memblock_alloc_try_nid_raw(size, align, max_addr,
diff --git a/mm/memblock.c b/mm/memblock.c
index 047947c9ed8d..e1fd07166a35 100644
--- a/mm/memblock.c
+++ b/mm/memblock.c
@@ -1412,11 +1412,9 @@ phys_addr_t __init memblock_alloc_range_nid(phys_addr_t size,
  * Return:
  * Physical address of allocated memory block on success, %0 on failure.
  */
-phys_addr_t __init memblock_alloc_range_nid_flags(phys_addr_t size,
-					phys_addr_t align, phys_addr_t start,
-					phys_addr_t end, int nid,
-					bool exact_nid,
-					enum memblock_flags flags)
+static phys_addr_t __init memblock_alloc_range_nid_flags(
+	phys_addr_t size, phys_addr_t align, phys_addr_t start, phys_addr_t end,
+	int nid, bool exact_nid, enum memblock_flags flags)
 {
 	phys_addr_t found;
 
-- 
2.17.1

